pkg/scheduler/queue/queue_test.go

```diff
@@ -63,10 +78,149 @@ func makeSchedulerRequest(tenantID string) *SchedulerRequest {
 			},
 			Url: "/prometheus/api/v1/query_range?end=1701720000&query=rate%28go_goroutines%7Bcluster%3D%22docker-compose-local%22%2Cjob%3D%22mimir-microservices-mode%2Fquery-scheduler%22%2Cnamespace%3D%22mimir-microservices-mode%22%7D%5B10m15s%5D%29&start=1701648000&step=60",
 		},
-		AdditionalQueueDimensions: randAdditionalQueueDimension(),
+		AdditionalQueueDimensions: additionalQueueDimensions,
+		EnqueueTime:               time.Now(),
 	}
 }
 
+// TestMultiDimensionalQueueFairnessSlowConsumerEffects emulates a simplified queue slowdown scenario
+// which the scheduler's additional queue dimensions features are intended to solve for.
+//
+// In this scenario, one category of queue item causes the queue consumer to slow down, introducing a
+// significant delay while the queue consumer processes it and before the consumer can dequeue the next item.
+// This emulates a situation where one of the query components - the ingesters or store-gateways - is under load.
+//
+// If queue items belonging to the slow category are in the same queue in front of the normal queue items,
+// the normal queue items must wait for all slow queue items to be cleared before they can be serviced.
+// In this way, the degraded performance of the slow query component equally degrades the performance of the
+// queries which *could* be serviced quickly, but are waiting behind the slow queries in the queue.
+//
+// With the additional queue dimensions enabled, the queues are split by which query component the query will utilize.
+// The queue broker then round-robins between the split queues, which has the effect of alternating between
+// dequeuing the slow queries and normal queries rather than blocking normal queries behind slow queries.
+func TestMultiDimensionalQueueFairnessSlowConsumerEffects(t *testing.T) {
+	promRegistry := prometheus.NewPedanticRegistry()
+
+	maxQueriersPerTenant := 0 // disable shuffle sharding
+	forgetQuerierDelay := time.Duration(0)
+	maxOutstandingRequestsPerTenant := 1000
+
+	totalRequests := 100
+	numTenants := 1
+	numProducers := 10
+	numConsumers := 1
+
+	normalQueueDimension := storeGatewayQueueDimension
+	slowConsumerLatency := 20 * time.Millisecond
+	slowConsumerQueueDimension := ingesterQueueDimension
```

Comment:
[nit] Using the store-gateway and ingester dimension names confused me for a second - can we just use names like these?

```suggestion
	normalQueueDimension := "normal-request"
	slowConsumerLatency := 20 * time.Millisecond
	slowConsumerQueueDimension := "slow-request"
```

Reply:
I agree, done !